k4MarlinWrapper/src/components/EDM4hep2Lcio.cpp

```diff
@@ -399,13 +485,39 @@ StatusCode EDM4hep2LcioTool::convertCollections(lcio::LCEventImpl* lcio_event) {
   }
   debug() << "Event: " << lcio_event->getEventNumber() << " Run: " << lcio_event->getRunNumber() << endmsg;
 
-  // Deal with
   EDM4hep2LCIOConv::sortParticleIDs(pidCollections);
+  if (!m_podioDataSvc) {
+    DataObject* p;
+    StatusCode  code = m_eventDataSvc->retrieveObject("/Event" + k4FWCore::frameLocation, p);
+    if (code.isSuccess()) {
+      auto* frame = dynamic_cast<AnyDataWrapper<podio::Frame>*>(p);
+      edmEvent    = std::cref(frame->getData());
+    } else {
+      auto frame = podio::Frame{};
+      edmEvent   = frame;
+    }
+  }
   for (const auto& pidCollMeta : pidCollections) {
-    const auto algoId = attachParticleIDMetaData(lcio_event, edmEvent, pidCollMeta);
+    auto algoId = attachParticleIDMetaData(lcio_event, edmEvent.value(), pidCollMeta);
```

Comment:
Maybe this also needs a similar treatment as `convertObjectParameters` in k4EDM4hep2LcioConv? That would make it possible to simplify the code below a bit, but I am not sure if it's worth the effort.